<a href="https://colab.research.google.com/github/AjayUnique/EDA-playstore-app/blob/main/Ajay_Tiwari_playstore.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>


# **The Play Store apps data has enormous potential to drive app-making businesses to success. Actionable insights can be drawn for developers to work on and capture the Android market.**
# **Each app (row) has values for catergory, rating, size, and more. Another dataset contains customer reviews of the android apps.**
# **Explore and analyze the data to discover key factors responsible for app engagement and success.**

In [5]:
import pandas as pd
import numpy as np 
import matplotlib.pyplot as plt
import seaborn as sns 
%matplotlib inline


In [6]:
import warnings
warnings.filterwarnings("ignore")

Importing the dataset into google colab

In [7]:
playdf = pd.read_csv("/content/drive/MyDrive/almabetter /Capstone project /Project 1 playstore/Play Store Data.csv")

In [8]:
playdf.columns

Index(['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type',
       'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver',
       'Android Ver'],
      dtype='object')

In [9]:
playdf.head(5)

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [10]:
playdf.describe()

Rating
count  9367.000000
mean      4.193338
std       0.537431
min       1.000000
25%       4.000000
50%       4.300000
75%       4.500000
max      19.000000

In [11]:
playdf.describe(include='O').T

count unique                 top   freq
App             10841   9660              ROBLOX      9
Category        10841     34              FAMILY   1972
Reviews         10841   6002                   0    596
Size            10841    462  Varies with device   1695
Installs        10841     22          1,000,000+   1579
Type            10840      3                Free  10039
Price           10841     93                   0  10040
Content Rating  10840      6            Everyone   8714
Genres          10841    120               Tools    842
Last Updated    10841   1378      August 3, 2018    326
Current Ver     10833   2832  Varies with device   1459
Android Ver     10838     33          4.1 and up   2451

In [12]:
playdf.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


Converting Rating,Reviews,Size,Installs and Price columns into 'int' dtype 

In [13]:
playdf.isnull().sum()

App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 1
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64

'Rating' contains 1474 null values 

In [14]:
playdf['App'].duplicated().value_counts()

False    9660
True     1181
Name: App, dtype: int64

1181 Apps are duplicated entries in the dataset

# **Data Cleaning**

In [15]:
playdf=playdf.drop_duplicates('App')

In [16]:
playdf.shape

(9660, 13)

All duplicates values are removed from 'App' column

In [17]:
playdf.isna().sum()

App                  0
Category             0
Rating            1463
Reviews              0
Size                 0
Installs             0
Type                 1
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64

In [18]:
#To minimize data loss we will fill the null values under "Rating" column by mean through category wise

In [19]:
playdf['Rating'].fillna(playdf.groupby('Category')['Rating'].transform('mean'),inplace=True)     

In [20]:
playdf.isna().sum()

App               0
Category          0
Rating            0
Reviews           0
Size              0
Installs          0
Type              1
Price             0
Content Rating    1
Genres            0
Last Updated      0
Current Ver       8
Android Ver       3
dtype: int64

In [21]:
playdf.dropna(inplace=True)

In [22]:
playdf.isna().sum()

App               0
Category          0
Rating            0
Reviews           0
Size              0
Installs          0
Type              0
Price             0
Content Rating    0
Genres            0
Last Updated      0
Current Ver       0
Android Ver       0
dtype: int64


Conversion of 'Reviews' column from object dtype to numerical dtype

In [23]:
playdf['Reviews']=playdf['Reviews'].astype(int)                   # converting into int dtype

In [24]:
playdf['Reviews'].dtype

dtype('int64')

Conversion of 'Size' column from object dtype to numerical dtype




In [25]:
playdf['Size'].unique()                              # Values contains M or k at the end as Megabytes or kilobytes resp.
                                                     #Some value contains 'Varies with devices' '

array(['19M', '14M', '8.7M', '25M', '2.8M', '5.6M', '29M', '33M', '3.1M',
       '28M', '12M', '20M', '21M', '37M', '5.5M', '17M', '39M', '31M',
       '4.2M', '7.0M', '23M', '6.0M', '6.1M', '4.6M', '9.2M', '5.2M',
       '11M', '24M', 'Varies with device', '9.4M', '15M', '10M', '1.2M',
       '26M', '8.0M', '7.9M', '56M', '57M', '35M', '54M', '201k', '3.6M',
       '5.7M', '8.6M', '2.4M', '27M', '2.7M', '2.5M', '16M', '3.4M',
       '8.9M', '3.9M', '2.9M', '38M', '32M', '5.4M', '18M', '1.1M',
       '2.2M', '4.5M', '9.8M', '52M', '9.0M', '6.7M', '30M', '2.6M',
       '7.1M', '3.7M', '22M', '7.4M', '6.4M', '3.2M', '8.2M', '9.9M',
       '4.9M', '9.5M', '5.0M', '5.9M', '13M', '73M', '6.8M', '3.5M',
       '4.0M', '2.3M', '7.2M', '2.1M', '42M', '7.3M', '9.1M', '55M',
       '23k', '6.5M', '1.5M', '7.5M', '51M', '41M', '48M', '8.5M', '46M',
       '8.3M', '4.3M', '4.7M', '3.3M', '40M', '7.8M', '8.8M', '6.6M',
       '5.1M', '61M', '66M', '79k', '8.4M', '118k', '44M', '695k', '1.6M',
     

In [26]:
playdf['Size'].replace('Varies with device',np.nan, inplace=True)             # converting 'Varies with device' value to Null value

In [27]:
playdf['Size']=playdf['Size'].apply(lambda x:str(x).replace('M',''))
playdf['Size']=playdf['Size'].apply(lambda x:str(x).replace('k','e-3'))       # replacing k(kilobytes) into Megabytes format 

In [28]:
playdf['Size'].shape

(9648,)

In [29]:
playdf['Size']=playdf['Size'].astype(float)

In [30]:
playdf['Size'].dtype

dtype('float64')

In [31]:
playdf['Size'].fillna(playdf.groupby('Category')['Size'].transform('mean'), inplace=True)

Conversion of 'Installs' column from object dtype to numerical dtype

In [32]:
playdf['Installs']

0            10,000+
1           500,000+
2         5,000,000+
3        50,000,000+
4           100,000+
            ...     
10836         5,000+
10837           100+
10838         1,000+
10839         1,000+
10840    10,000,000+
Name: Installs, Length: 9648, dtype: object

In [33]:
playdf['Installs']=playdf['Installs'].apply(lambda x:x.replace('+',''))
playdf['Installs']=playdf['Installs'].apply(lambda x:x.replace(',',''))

In [34]:
playdf['Installs']=playdf['Installs'].astype(int)                   # converting into int dtype

In [35]:
playdf['Installs'].dtype

dtype('int64')

In [36]:
#Removing rows in which Reviews is greater than Installs. 

conditions = [    playdf['Reviews']>playdf['Installs'],                
                  playdf['Reviews']<=playdf['Installs']          ]

choice = ['Remove', 'Keep']

playdf['Decision'] = np.select(conditions, choice)

In [37]:
playdf['Decision'].value_counts()

Keep      9637
Remove      11
Name: Decision, dtype: int64

In [38]:
playdf.shape

(9648, 14)

In [39]:
playdf['Price']=playdf['Price'].apply(lambda x:x.replace('$',''))         # converting dtype of 'Price' from object to numerical.

In [40]:
playdf['Price'].astype(float)

0        0.0
1        0.0
2        0.0
3        0.0
4        0.0
        ... 
10836    0.0
10837    0.0
10838    0.0
10839    0.0
10840    0.0
Name: Price, Length: 9648, dtype: float64

In [41]:
playdf['Price']=playdf['Price'].astype(float)


In [42]:
playdf.isna().sum()

App               0
Category          0
Rating            0
Reviews           0
Size              0
Installs          0
Type              0
Price             0
Content Rating    0
Genres            0
Last Updated      0
Current Ver       0
Android Ver       0
Decision          0
dtype: int64

In [43]:
playdf.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 9648 entries, 0 to 10840
Data columns (total 14 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             9648 non-null   object 
 1   Category        9648 non-null   object 
 2   Rating          9648 non-null   float64
 3   Reviews         9648 non-null   int64  
 4   Size            9648 non-null   float64
 5   Installs        9648 non-null   int64  
 6   Type            9648 non-null   object 
 7   Price           9648 non-null   float64
 8   Content Rating  9648 non-null   object 
 9   Genres          9648 non-null   object 
 10  Last Updated    9648 non-null   object 
 11  Current Ver     9648 non-null   object 
 12  Android Ver     9648 non-null   object 
 13  Decision        9648 non-null   object 
dtypes: float64(3), int64(2), object(9)
memory usage: 1.1+ MB


Now we have five numerical dtype columns such as 'Rating ,Reviews, Size, Installs and Price'


---

